docs/source/cpp_api/neighbors_ball_cover.rst

```diff
@@ -0,0 +1,19 @@
+Random Ball Cover
+=================
+
+This page provides C++ class references for the publicly-exposed elements of the neighbors package.
```

Comment:
Was wondering, for all cases, why is it necessary to repeat this string in all the sub `.rst` files of each group?

Reply:
It's being copied for now only so that the pages are consistent, but I have it as a personal TODO to go through these and customize each brief heading description for both the top-level and subsections. (I could keep adding to this PR but it's so massive already).

Reply:
I just removed the headings from all the subsections.